src/app/Main/Body/QueryPage/Results/TableResults/TableResults.model.ts

```diff
@@ -1,4 +1,9 @@
 export interface SumOptions {
     showRowSum: boolean,
     showColSum: boolean
-  }
\ No newline at end of file
+  }
+
+export interface DisplayRowCol {

```

Comment:
As i commented, i do not really understand what that flags do. So there should be a better name for that interface (and the component's props) making things more clear.
In general an interface with more than one property should not be named with a verb but with a name describing the data the interface encapsulates. So `SumOptions` is a good name, `DisplayRowCol` not.
Also names for exported interfaces in your model files (names for your component props too) should be selected carefully, these are part of the code another developer would look at first to understand your code and the underlying structure of the data.

Reply:
What would you suggest?

Reply:
Hehe, i would have given a suggestion if i had a good one.
So `DisplayOptions` would be to general, `DataConfiguration` is a little bit better. 
In general you should handle your table more like a multi-dimensional table with two dimensions (as we discussed also for the api), because the type of data in rows and cols is the same, so there are no rows representing entries and cols representing attributes of those entries. Your app data should be represented with such a structure and your table specific data should be data independent and just represent headers, rows, cols and the cells.
The clearest solution would be to represent the data configuration like:
```
interface IsolateDataFormat {
  rowAttribute: IsolateAttribute | null,
  colAttribute: IsolateAttribute | null
```
and change the attributes accordingly to the user input to the table options (data to show in row, col). And then derive the  table data from this in the results container: 
```
  if(dataFormat.rowAttribute !== null) {
    addRowHeaders(dataFormat.rowAttribute)
  } else if(...)
```
So in the end isCol and isRow are derived states that do not need to be modeled explicitly (but they can still be used as local variables).

Reply:
Like this? Commit: "Check isCol?|isRow? local (no prop)"

Reply:
Yes, you did exactly what i thought of. Kind of overhead for just a renaming proposal but it makes the code better.
If you have the time (outside this ticket) you could dive into two things to make the row/col !== null checks more convenient:
First you could check if react proposes some data selection functionality. I think the reducer mechanism should have something like it. The idea here is that you store your app data via reducers and access it via selectors. The selectors are just a kind of function to get peaces of the data in the required form. Something like: `selectIsRowDataDefined(tableData)`. If not you can define your own getter functions beside the model definition like:
```
export interface TableData ...
export function getIsRowDataDefined(tableData: TableData): boolean {...}
```
Second, you could check if it is possible to use classes with react state. This would be the more object-oriented encapsulated way. Then you could define your tableData as a class with integrated accessors:
```
export class TableData {
  ...
  public get isRowDataDefined(): boolean { return this.row !== undefined }
}
```
So the idea is to find a way to handle derived data with react. For our angular projects we use a store mechanism with reducers and selectors.

Reply:
Make a new ticket for row/col !== null. Using classes with React state is not recommended.